app/src/main/java/org/wikipedia/page/PageTitle.kt

```diff
@@ -32,7 +32,8 @@ data class PageTitle(
     var description: String? = null,
     // TODO: remove after the restbase endpoint supports ZH variants.
     @SerialName("displayText") private var _displayText: String? = null,
-    var extract: String? = null
+    var extract: String? = null,
+    var redirectedFrom: String? = null,
```

Comment:
Adding a `redirectedFrom` field here is not a clean way of passing around this information. It should be easy enough to pass the redirected name directly from `PageFragmentLoadState` into `onPageMetadataLoaded()`.
(And don't worry about preserving the `redirectedFrom` name from the link preview dialog.)

Reply:
Thanks for the feedback! I have moved the `redirectedFrom` parameter to the `onPageMetadataLoaded(...)` method. Now the app only shows the redirected snackbar when clicking a Wikipedia link in an external application.

Reply:
> Adding a `redirectedFrom` field here is not a clean way of passing around this information. It should be easy enough to pass the redirected name directly from `PageFragmentLoadState` into `onPageMetadataLoaded()`. (And don't worry about preserving the `redirectedFrom` name from the link preview dialog.)